src/main/java/ClientConnection.java

```diff
@@ -16,7 +16,7 @@ public ClientConnection(int portNumber) throws IOException {
     server = new ServerSocket(portNumber);
   }
 
-  public String receiveRequest() throws IOException {
+  public String[] receiveRequest() throws IOException {
```

Comment:
I had made a StringParser class at first but then deleted it as it only had the one method that performed one line of code, the split(). I went ahead and rewrote that class and changed ConnectionClient back to the way it was, and updated the HttpServerRunner to use this new class.